### PR TITLE
Updated chronjob

### LINKED ISSUE #21: cronjob handling
Hello,
why are you writing the cronjob into sudos crontab,
can't you just handle the cronjobs in a new file in `/etc/cron.d`?
(e. g. https://unix.stackexchange.com/a/296351)

### PR DESCRIPTION
**Pull request by Targunitoth**
Update install.sh to create cronjobs in /etc/cron.d/
Fixes #21

### CODE REVIEW

**Comment by Targunitoth** [comment]
let the user choose the update frequency is a suggestion for #21

File: install.sh [hunk 1]
```diff
@@ -114,20 +115,53 @@ EOF
                 echo "[i] File created."
 
                 # add cronjob
-                read -r -p "[?] Create a cronjob/systemd-service which updates /etc/hosts with new adservers once a week? [systemd/cronjob/N] " REPLY
+                read -r -p "[?] Create a cronjob/systemd-service which updates /etc/hosts with new adservers? [systemd/cronjob/N] " REPLY
                 case "${REPLY}" in
                     [Cc][Rr][Oo][Nn][Jj][Oo][Bb] | [Cr][Rr][Oo][Nn][Tt][Aa][Bb] | [Cc][Rr][Oo][Nn] | [Cc] ) # CRONJOB, CRONTAB, CRON, C
+                        read -r -p "[?] How often should the cronjob run? [weekly/DAILY/hourly/reboot] " FREQUENCY
+                        # set daily as default
+                        line="1 12 * * * root ${SCRIPT_DIR}/adaway-linux.sh"
```
Comment (sedrubal) [inline install.sh]: Can you set only the cron time expression in the switch case and add the `root ${SCRIPT_DIR}/adaway-linux.sh` only once for all cases in order to [not repeat yourself](https://en.wikipedia.org/wiki/Don't_repeat_yourself)?

File: install.sh [hunk 2]
```diff
@@ -114,20 +115,53 @@ EOF
...
+                        esac
                         echo "[i] Creating cronjob..."
-                        line="1 12 */5 * * ${SCRIPT_DIR}/adaway-linux.sh"
-                        (crontab -u root -l; echo "${line}" ) | crontab -u root -
+                        echo "${line}" > ${CRONJOB_FILE}
+                        # make sure permissions are right
+                        chmod u=rw,g=r,o=r ${CRONJOB_FILE}
+                        chown root:root ${CRONJOB_FILE}
+                        # restart the cron service
+                        service cron restart
```
Comment (sedrubal) [inline install.sh]: Do we really need to restart a service? I tested on fedora and there is neither a cron service nor you have to restart anything. Additionally using service is deprecated on most distributions that use systemd.

**Comment by Targunitoth** [comment]
Please check this request.
The "cron" service in Fedora should be named "crond".
Loading changes from /etc/cron.d/ needs a service restart, a cron reload is not enougth.

**Comment by sedrubal** [comment]
Ah yes it is called crond of course. But at least for crond on fedora and debian you don't need to restart the deamon because it seems to use the modtime option (see `man crond(8)`). If you really think, we need it (with all the overhead), I'm ok with your changes.

Another thing: Do you really think, we should update the hosts file every day? I've never monitored how often the providers update their lists, but I think there won't be large changes within one day...

**Comment by Targunitoth** [comment]
You are right, `/etc/cron.d` will be be updated automatically. The problem is, that the man-page is not really elaborately. I knew it reloads the users `/var/spool/cron`, but I was not sure about `/etc/cron.d`. 
In my first test cron failed to reload and execute the `adaway-linux.sh`. So I read in a forum, that `sudo service cron reload` is not enouth for `/etc/cron.d`, you have to restart the whole cron daemon.
I don't know, what I did wrong, but now it is working without restarting and I have no problems.
Therefore I removed the overhead.

To the Update frequency: The ad-companies and the hosts-file-provider are playing a cat-and-mouse game. It is not hard to create a new ad-domain, so there are constant changes. But there are also often updates on the hosts-files: `hosts-file.net` updated at 22nd of March and `pgl.yoyo.prg` updated yesterday (at 25th of March). So a daily update has a right to exist. 

This is your Program, if you want to set the default update frequency to weekly I have no problem with this at all. As long as I can choose my personal  update frequency.

**Comment by stablestud** [comment]
> As long as I can choose my personal update frequency.

How about a global option that'll set the update frequency to whatever the user prefers?
That would be a neat implementation. Feel free to add it if you want.

**Comment by Targunitoth** [comment]
The option is already in my pull-request.
We just have to negotiate the default value.
@stablestud What would you prefer? Weekly or daily updates?

**Comment by sedrubal** [comment]
With your example above I'm convinced. I'll merge this asap (currently I don't have time).

**Comment by stablestud** [comment]
@Targunitoth I personally prefer weekly, however this is only my personal preference, as I have a laptop and it would cost too much valuable bandwidth when I'm using it mobile.  BTW: good idea to be able choose the update frequency. :+1:

Note: You may want to enclose all your variables `${var}` with `"` like so: `"${var}"`

**Comment by Targunitoth** [comment]
Thank you for the hint. 
`${CRONJOB_FILE}` is a readonly system variable and should not contain any spaces. But I changed it.
`${freq}` is inside the cat command, there are no `"` _allowed_

**Pull request merged**